tests/sentry/auth/test_superuser.py

```diff
@@ -189,6 +201,33 @@ def test_su_access_no_request(self, logger):
                 "superuser.logged-in", extra={"ip_address": "127.0.0.1", "user_id": 10}
             )
 
+    @freeze_time(BASETIME + INSIDE_PRIVILEGE_ACCESS_EXPIRE_TIME, as_arg=True)
+    def test_not_expired_check_org_in_request(frozen_time, self):
+        request = self.build_request()
+        superuser = Superuser(request, allowed_ips=())
+        assert superuser.is_active is True
+
+        frozen_time.move_to(BASETIME + OUTSIDE_PRIVILEGE_ACCESS_EXPIRE_TIME)
+
+        assert superuser.is_active is True
```

Comment:
do you have to re-initialize the superuser object here? or is is_active calculated again?

Reply:
every time the superuser object is called, the idle expiry time is refreshed,  if i didn't call the superuser object on line 208, it would expire and line 212 would be false as it would surpass the idle expiry time

Reply:
@maxiuyuan when you say called, what do you mean? you're saying that accessing `is_active` will refresh the expiry time?

Reply:
not accessing is_active specifically. `on_response` will update the idle expiration window, which will be called in the middleware, which to my understanding, will be every time the superuser object is accessed